src/util/Combinatorics.java

```diff
@@ -0,0 +1,39 @@
+package util;
+
+import java.util.*;
+
+public class Combinatorics {
+
+    public static List<Set> allCombinationsOf(Set objects) {
```

Comment:
Could it be made a [generic method](https://docs.oracle.com/javase/tutorial/extra/generics/methods.html)?